modules/cnf-performing-platform-verification-latency-tests.adoc

```diff
@@ -0,0 +1,756 @@
+
+|`MAXIMUM_LATENCY`
+|This is a unified variable you can apply for all the available latency tools.
+
+|====
+
+[NOTE]
+====
+A variable that is specific to certain tests has precedence over the unified variable.
+====
+
+[id="cnf-performing-end-to-end-tests-running-the-tests_{context}"]
+== Running the latency tests
+
+Run the latency tests against the cluster by completing the following steps:
+
+. Open a shell prompt in the directory containing the `kubeconfig` file.
++
+You provide the test image with a `kubeconfig` file in current directory and its related `$KUBECONFIG` environment variable, mounted through a volume. This allows the running container to use the `kubeconfig` file from inside the container.
+
+. Run the latency tests by entering the following command:
++
+[source,terminal,subs="attributes+"]
+----
+$ podman run -v $(pwd)/:/kubeconfig:Z -e KUBECONFIG=/kubeconfig/kubeconfig \
+-e DISCOVERY_MODE=true registry.redhat.io/openshift4/cnf-tests-rhel8:v{product-version} \
+/usr/bin/test-run.sh -ginkgo.focus="\[performance\]\ Latency\ Test"
+----
++
+[IMPORTANT]
+====
+You must run the latency tests with `-e DISCOVERY_MODE=true`. The latency tests can change the configuration of your cluster if you do not run in Discovery mode.
+====
+
+. Optional: Append `-ginkgo.dryRun` to run the latency tests in dry-run mode. This is useful for checking what the tests run.
+
+. Optional: Append `-ginkgo.v` to run the tests with increased verbosity.
+
+. Optional: To run the latency tests against a specific performance profile, run the following command, substituting appropriate values:
++
+[source,terminal,subs="attributes+"]
+----
+$ podman run -v $(pwd)/:/kubeconfig:Z -e KUBECONFIG=/kubeconfig/kubeconfig \
+-e LATENCY_TEST_RUN=true -e LATENCY_TEST_RUNTIME=600 -e OSLAT_MAXIMUM_LATENCY=20 \
+-e PERF_TEST_PROFILE=<performance_profile> registry.redhat.io/openshift4/cnf-tests-rhel8:v{product-version} \
+/usr/bin/test-run.sh -ginkgo.focus="\[performance\]\[config\]|\[performance\]\ Latency\ Test"
+----
++
+where:
++
+--
+<performance_profile> :: Is the name of the performance profile you want to run the latency tests against.
+--
+
+[id="cnf-performing-end-to-end-tests-running-hwlatdetect"]
+=== Running hwlatdetect
+
+The `hwlatdetect` tool is available in the `rt-kernel` package with a regular subscription of Red Hat Enterprise Linux 8.
+
+.Prerequisites:
+
+* You have installed the real-time kernel in the cluster
+
+* You have logged in to `registry.redhat.io` with your Customer Portal credentials
+
+.Procedure
+
+* To run the `hwlatdetect` tests, run the following command, substituting variable values as appropriate:
++
+[source,terminal,subs="attributes+"]
+----
+$ podman run -v $(pwd)/:/kubeconfig:Z -e KUBECONFIG=/kubeconfig/kubeconfig \
+-e LATENCY_TEST_RUN=true -e DISCOVERY_MODE=true -e ROLE_WORKER_CNF=worker-cnf \
+-e LATENCY_TEST_CPUS=600 -e LATENCY_TEST_RUNTIME=600 \
```

Comment:
600 is not a valid value for LATENCY_TEST_CPUS. this is an optional env var, you can remove it from the command.

Reply:
also, it doesn't match the output, the output matches a command that LATENCY_TEST_CPUS was not specified in it.